src/lib/seam/connect/seam-http-request.ts

```diff
@@ -0,0 +1,115 @@
+import type { Method } from 'axios'
+
+import type { Client } from './client.js'
+import { SeamHttpActionAttempts } from './index.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { resolveActionAttempt } from './resolve-action-attempt.js'
+
+export interface SeamHttpRequestParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+export interface SeamHttpRequestConfig<TBody> {
+  url?: string
+  method?: Method
+  params?: any
+  data?: TBody
+}
+
+export type ResponseFromSeamHttpRequest<T> =
+  T extends SeamHttpRequest<any, infer TResponse, infer TResourceKey>
+    ? TResourceKey extends keyof TResponse
+      ? TResponse[TResourceKey]
+      : undefined
+    : never
+
+export class SeamHttpRequest<
+  const TBody,
+  const TResponse,
+  const TResourceKey extends keyof TResponse | undefined,
```

Comment:
Can we put TResponse and TBody last and default them? I know it won't change the generated code, but it reads more logically	
```suggestion
  const TResourceKey extends keyof TResponse | undefined,
  const TBody = undefined,
  const TResponse = void,
```

Reply:
I'm not sure that I agree that it reads more naturally - we have the body that we're sending, the response that we're getting and the key that we're plucking out of that response. Maybe `TResourceKey` should be `TResponseKey` though.

Reply:
> Maybe TResourceKey should be TResponseKey though.
Sure!

And, yea, I order is not so important, I was thinking these were request related but they are all optional anyway, so nevermind my comment.